components/director/internal/domain/packageinstanceauth/service.go

```diff
@@ -67,6 +94,50 @@ func (s *service) GetForPackage(ctx context.Context, id string, packageID string
 	return pkg, nil
 }
 
+func (s *service) List(ctx context.Context, packageID string) ([]*model.PackageInstanceAuth, error) {
+	tnt, err := tenant.LoadFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	pkgInstanceAuths, err := s.repo.ListByPackageID(ctx, tnt, packageID)
+	if err != nil {
+		return nil, errors.Wrap(err, "while listing Package Instance Auths")
+	}
+
+	return pkgInstanceAuths, nil
+}
+
+func (s *service) SetAuth(ctx context.Context, id string, in model.PackageInstanceAuthSetInput) error {
+	tnt, err := tenant.LoadFromContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	instanceAuth, err := s.repo.GetByID(ctx, tnt, id)
+	if err != nil {
+		return errors.Wrap(err, "while getting Package Instance Auth")
+	}
+	if instanceAuth == nil {
```

Comment:
Shouldn't the repo return `notFoundError`?

Reply:
It does, I added this check to avoid panics in case `nil, nil` is somehow returned.